### PR TITLE
docs: service-registry-doc

### PR DESCRIPTION
**Pull request by maancham**
## Description

AXE-3150

## Todos

- [ ] Unit tests
- [ ] Manual tests
- [ ] Documentation
- [ ] Connect epics/issues

### CODE REVIEW

File: doc/src/contracts/service_registry.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# Service Registry
+
+The service registry keeps track of the pool of workers that are related to each service.
```
Comment (cjcobb23) [inline doc/src/contracts/service_registry.md:3]:
```suggestion
The service registry keeps track of the pool of workers that vote and sign for each chain.
```

File: doc/src/contracts/service_registry.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+The core functionalities, such as registering a new service, worker authorization and un-authorization can only be called 
+from a governance address. To become active and be eligible to participate in voting for a specific service,
```
Comment (cjcobb23) [inline doc/src/contracts/service_registry.md:5]:
```suggestion
from a governance address. Worker bonding and unbonding, as well as registering support for specific chains, are called by the worker themselves. To become active and be eligible to participate in voting for a specific chain, 
```

File: doc/src/contracts/service_registry.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
+# Service Registry
+
+The service registry keeps track of the pool of workers that are related to each service.
+The core functionalities, such as registering a new service, worker authorization and un-authorization can only be called 
+from a governance address. To become active and be eligible to participate in voting for a specific service, 
+each worker should first be authorized and the bond enough stake. Service registry is used by ampd, verifier, and multisig prover.
```
Comment (cjcobb23) [inline doc/src/contracts/service_registry.md:6]:
```suggestion
each worker should first be authorized and bond enough stake, and then register support for that chain. 

Service registry is used by ampd, verifier, and multisig prover.
```

File: doc/src/contracts/service_registry.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+
```
Comment (cjcobb23) [inline doc/src/contracts/service_registry.md:8]: Can we add a paragraph here talking about what exactly is a service? Versus a chain. Call out specifically that we plan to use one service, and all chains are within that service.

File: doc/src/contracts/service_registry.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    Vr{"Voting Verifier"}
+    Pr{"Prover"}
+    R{"Service Registry"}
+end
+OC{"Workers"}
+
+Vr -- "GetActiveWorkers" --> R
+Pr -- "GetActiveWorkers" --> R
+R -- "Vec[Worker]" --> Vr
+R -- "Vec[Worker]" --> Pr
```
Comment (cjcobb23) [inline doc/src/contracts/service_registry.md:75]: I think for clarity we can remove these lines

File: doc/src/contracts/service_registry.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+OC -- "Vote(poll_id, votes)" --> Vr
```
Comment (cjcobb23) [inline doc/src/contracts/service_registry.md:76]: Remove, I don't think this is necessary

File: doc/src/contracts/service_registry.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+Governance->>+Service Registry: Register Service
+Governance->>+Service Registry: Authorize Workers
+
+
+Worker->>+Service Registry: Register Chain Support
+Worker->>+Service Registry: Bond Worker
+Worker->>+Service Registry: Claim Stake
+
+
+Service Registry->>Worker: returns BankMsg with amount set to released_bond
```
Comment (cjcobb23) [inline doc/src/contracts/service_registry.md:103]: Let's remove the claiming stake part here. I'd rather the sequence diagram just show the worker onboarding process. Claiming stake is off boarding. Also, the worker actually needs to unbond first, before claiming stake. If you want, we can add a separate diagram for off boarding. Claiming stake requires waiting for some unbonding period.

File: doc/src/contracts/service_registry.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+end
+actor Governance
+actor Worker
+
+Governance->>+Service Registry: Register Service
+Governance->>+Service Registry: Authorize Workers
+
+
+Worker->>+Service Registry: Register Chain Support
+Worker->>+Service Registry: Bond Worker
```
Comment (cjcobb23) [inline doc/src/contracts/service_registry.md:99]: Can we swap the order of these?

File: doc/src/contracts/service_registry.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+Worker->>+Service Registry: Claim Stake
+
+
+Service Registry->>Worker: returns BankMsg with amount set to released_bond
+```
+
+1. The Governance registers a new service by providing the necessary parameters for the service.
+2. Governance is also responsible for authorizing workers to join the service by sending an `Authorize Workers` message.
```
Comment (cjcobb23) [inline doc/src/contracts/service_registry.md:107]: Can we add something saying authorizing and bonding can be done in any order?

File: doc/src/contracts/service_registry.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+3. Workers register support for specific chains within the service by specifying service name and chain names.
+4. Workers bond to the service, providing stake, by sending a `Bond Worker` message with appropriate funds denominator.
+5. Workers can claim released stake with `Claim Stake` message, and the Service Registry responds with a BankMsg containing the released bond.
+
+
```
Comment (cjcobb23) [inline doc/src/contracts/service_registry.md:112]: We probably also want to add something that says workers need to register their public key to participate in signing, and add a link to the multisig contract documentation.

**Review by cjcobb23 — CHANGES_REQUESTED** [review]

File: doc/src/contracts/service_registry.md (hunk 4 above)
Comment (maancham) [reply doc/src/contracts/service_registry.md:8]: added

File: doc/src/contracts/service_registry.md (hunk 7 above)
Comment (maancham) [reply doc/src/contracts/service_registry.md:103]: I removed the lines

File: doc/src/contracts/service_registry.md (hunk 8 above)
Comment (maancham) [reply doc/src/contracts/service_registry.md:99]: done

File: doc/src/contracts/service_registry.md (hunk 10 above)
Comment (maancham) [reply doc/src/contracts/service_registry.md:112]: I added a Notes part

File: doc/src/contracts/service_registry.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+To become active and be eligible to participate in voting for a specific chain,
+each worker should first be authorized and bond enough stake, and then register support for that chain.
+Service registry is used by ampd, verifier, and multisig prover.
+
+The term `service` refers to an upper-level entity that includes several 
+`chains`. The difference is in how they are related to each other, which is 
+hierarchical; a service is like an umbrella that regulates the activities of 
+several chains that fall under its purview. The service defines common 
+parameters, such as worker requirements, bonding details, and unbonding periods, which are applicable to all associated chains. 
+Thus, we use a single instance of service registry to organize and coordinate activities across all chains.
```
Comment (cjcobb23) [inline doc/src/contracts/service_registry.md:15]: Nice summary!

**Review by cjcobb23 — APPROVED** [review]

**Pull request merged**